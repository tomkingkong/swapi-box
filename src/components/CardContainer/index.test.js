import React from "react";

import { CardContainer } from "./index";

describe("CardContainer", () => {
  let wrapper;
  let mockData;
  let toggleFavorites;
  let savedFavorites;
  let handlePage;

  beforeEach(() => {
    mockData = [
      {
        name: "Luke Skywalker",
        species: "Human",
        homeworld: "Tatooine",
        population: "200000",
        favorite: false
      }
    ];
    toggleFavorites = jest.fn();
    savedFavorites = [];
    handlePage = jest.fn();

    wrapper = shallow(
      <CardContainer
        data={mockData}
        handlePage={handlePage}
        toggleFavorites={toggleFavorites}
        savedFavorites={savedFavorites}
        isFavorites={false}
      />
    );
  });

  it("should match snapshot with data object passed", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot without data object passed", () => {
    mockData = null;
    wrapper = shallow(
      <CardContainer
        data={mockData}
        handlePage={handlePage}
        toggleFavorites={toggleFavorites}
        savedFavorites={savedFavorites}
        isFavorites={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with favorites and data", () => {
    savedFavorites = ["Luke Skywalker"];
    wrapper = shallow(
      <CardContainer
        data={mockData}
        handlePage={handlePage}
        toggleFavorites={toggleFavorites}
        savedFavorites={savedFavorites}
        isFavorites={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with favorites and no data", () => {
    savedFavorites = ["Luke Skywalker"];
    wrapper = shallow(
      <CardContainer
        data={[]}
        handlePage={handlePage}
        toggleFavorites={toggleFavorites}
        savedFavorites={savedFavorites}
        isFavorites={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handlePage on click", () => {
    let mockHandlePage = jest.fn();

    wrapper = shallow(
      <CardContainer
        data={mockData}
        handlePage={mockHandlePage}
        toggleFavorites={toggleFavorites}
        savedFavorites={savedFavorites}
      />
    );
    wrapper
      .find(".prev-page-btn")
      .first()
      .simulate("click");
    expect(mockHandlePage).toHaveBeenCalled();

    wrapper
      .find(".next-page-btn")
      .first()
      .simulate("click");
    expect(mockHandlePage).toHaveBeenCalled();
  });
});
/* global shallow */