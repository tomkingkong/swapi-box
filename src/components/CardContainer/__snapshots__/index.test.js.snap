// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot with data object passed 1`] = `
<section
  className="card_container"
>
  <div
    className="buttons"
  >
    <button
      className="prev-page-btn"
      onClick={[Function]}
    >
      Prev
    </button>
    <button
      className="next-page-btn"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
  <div
    className="center_card_container"
  >
    <Component
      card={
        Object {
          "favorite": false,
          "homeworld": "Tatooine",
          "name": "Luke Skywalker",
          "population": "200000",
          "species": "Human",
        }
      }
      key="0"
      toggleFavorites={[Function]}
    />
  </div>
</section>
`;

exports[`CardContainer should match snapshot with favorites and data 1`] = `
<section
  className="favorites card_container"
>
  <div
    className="buttons"
  >
    <button
      className="prev-page-btn"
      onClick={[Function]}
    >
      Prev
    </button>
    <button
      className="next-page-btn"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
  <div
    className="center_card_container"
  >
    <Component
      card={
        Object {
          "favorite": true,
          "homeworld": "Tatooine",
          "name": "Luke Skywalker",
          "population": "200000",
          "species": "Human",
        }
      }
      key="0"
      toggleFavorites={[Function]}
    />
  </div>
</section>
`;

exports[`CardContainer should match snapshot with favorites and no data 1`] = `
<section
  className="card_container"
>
  <p
    className="empty-fav"
  >
    These are not the favorites you are looking for...
  </p>
</section>
`;

exports[`CardContainer should match snapshot without data object passed 1`] = `
<section
  className="card_container loading"
>
  <img
    height="100px"
    src="https://media2.giphy.com/media/10MKHgkZMDlQ4M/giphy.gif"
    width="100px"
  />
</section>
`;
